Provide a 'Changelog' link on rubygems.org/gems/mime-types-data

mark-young-atg opened the pull request:
By providing a 'changelog_uri' in the metadata of the gemspec a 'Changelog' link will be shown on https://rubygems.org/gems/mime-types-data which makes it quick and easy for someone to check on the changes introduced with a new version.

Details of this functionality can be found on https://guides.rubygems.org/specification-reference/#metadata

halostatue reviewed (changes requested):
`mime-types-data.gemspec` is a generated file.

The changelog will be added to the generated gemspec if you update the `README` to include the following line in `README.md` with the other links.

```markdown
- changelog :: https://github.com/mime-types/mime-types-data/blob/main/CHANGELOG.md
```

mark-young-atg commented:
@halostatue Thank you very much for the quick review and suggestion. Hopefully I've now incorporated that correctly into this PR. I reverted the `mime-types-data.gemspec` file to it's original contents, was that correct?

Cheers,
Mark

halostatue commented on .github/workflows/reviewdog.yml:
This file shouldn't be part of your change as it's already on `main`.

mark-young-atg replied:
Huh, I think was a dependabot made change that showed up in my PR. I didn't touch that file. I rebased against `main` and I think it looks right now.

halostatue reviewed (changes requested):
I will request a couple of other changes:

- Run `bundle && bundle exec rake gemspec` to update the `.gemspec`.
- Add yourself to the `CONTRIBUTORS.md`
- Add an entry describing the addition of the changelog link to the `CHANGELOG.md` under a new section (`## NEXT / YYYY-MM-DD`). This should also link to `[#96][pull-96]` and add `[pull-96]: https://github.com/mime-types/mime-types-data/pull/96` to the bottom of the file.

I’ll probably need to fix the automated update code (because it should detect the presence of a `NEXT / YYYY-MM-DD` and _replace it_), but that'll be a welcome change.

halostatue reviewed (approved):
Thank you very much for your contribution and your patience with my change requests.
